tests/test_service.py

```diff
@@ -106,13 +106,86 @@ def test_create_supplier_wrong_content_type(self):
+        post_resp = self.app.post('/suppliers', json=test_supplier, content_type='application/json')
+        
+        posted_data = post_resp.get_json()
+        resp = self.app.get("/suppliers/{}".format(posted_data['_id']), content_type="application/json")
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+        data = resp.get_json()
+
+        data['name'] = 'supplier2'
+ 
+        resp = self.app.put('/suppliers/{}'.format(data['_id']), json=data, content_type='application/json')
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+        
+        resp = self.app.get('/suppliers/{}'.format(data['_id']), content_type='application/json')
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+        data = resp.get_json()
+        logging.debug('data = %s', data)
+        self.assertEqual(data['name'], 'supplier2')
+    
+    
+    def test_update_supplier_with_no_name(self): 
+        new_supplier = {"id": 1, "name": "supplier1", "like_count":2, "is_active": True, "products": [1,2,3], "rating":8.5}
+        
+        post_resp = self.app.post('/suppliers', json=new_supplier, content_type='application/json')
+        posted_data = post_resp.get_json()
+        
+        resp = self.app.get("/suppliers/{}".format(posted_data['_id']), content_type="application/json")
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+
+        data = resp.get_json()
+        
+        del data['name']
+        resp = self.app.put('/suppliers/{}'.format(data['_id']), json=data, content_type='application/json')
+        self.assertEqual(resp.status_code, HTTP_400_BAD_REQUEST)
+ 
+    """
+    def test_update_supplier_with_wrong_id(self):
+        new_supplier = {"id": 1, "name": "supplier1", "like_count":2, "is_active": True, "products": [1,2,3], "rating":8.5}
+
+        post_resp = self.app.post('/suppliers', json=new_supplier, content_type='application/json')
+        posted_data = post_resp.get_json()
+
+        resp = self.app.get("/suppliers/{}".format(posted_data['_id']), content_type="application/json")
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+
+        data = resp.get_json()
+       
+        del data["like_count"]
+        resp = self.app.put('/suppliers/{}'.format(posted_data['_id']), json=data, content_type="application/json")
+        self.assertEqual(resp.status_code, HTTP_400_BAD_REQUEST)
+    """     
+
+    def test_update_supplier_not_found(self):
+        """Update a Supplier that does not exist"""
+        new_supplier = {"id": 1, "name": "supplier1", "like_count":2, "is_active": True, "products": [1,2,3], "rating": 8.5}
+        resp = self.app.put('/suppliers/0', json=new_supplier, content_type='application/json')
+        self.assertEqual(resp.status_code, HTTP_404_NOT_FOUND)
+
+
+####################################################:##################
 # Utility functions
 ######################################################################
+    def get_supplier(self, name):
```

Comment:
This function was provided by the prof. But was not working, so I wrote my own update test method. But just in case I left out that code there if we might need it in the future.